src/Microsoft.DotNet.CodeFormatting/Analyzers/ExplicitVariableTypeAnalyzer.cs

```diff
@@ -0,0 +1,81 @@
+using Microsoft.CodeAnalysis.Diagnostics;
+using System.Diagnostics;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.CodeFormatting.Analyzers
+{
+    [Export(typeof(DiagnosticAnalyzer))]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    class ExplicitVariableTypeAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "DNS0099";
+        internal const string VariableDeclarationCustomTag = "VariableDeclarationTag";
+        internal const string ForEachStatementCustomTag = "ForEachStatementTag";
+
+        private static DiagnosticDescriptor ruleVariableDeclaration = new DiagnosticDescriptor(DiagnosticId,
+                                                                            ResourceHelper.MakeLocalizableString(nameof(Resources.ExplicitVariableTypeAnalyzer_Title)),
+                                                                            ResourceHelper.MakeLocalizableString(nameof(Resources.ExplicitVariableTypeAnalyzer_MessageFormat)),
+                                                                            "Style",
+                                                                            DiagnosticSeverity.Warning,
+                                                                            true,
+                                                                            customTags: VariableDeclarationCustomTag);
+        private static DiagnosticDescriptor ruleForEachStatement = new DiagnosticDescriptor(DiagnosticId,
+                                                                            ResourceHelper.MakeLocalizableString(nameof(Resources.ExplicitVariableTypeAnalyzer_Title)),
+                                                                            ResourceHelper.MakeLocalizableString(nameof(Resources.ExplicitVariableTypeAnalyzer_MessageFormat)),
+                                                                            "Style",
+                                                                            DiagnosticSeverity.Warning,
+                                                                            true,
+                                                                            customTags: ForEachStatementCustomTag);
+
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> s_supportedRules = ImmutableArray.Create(ruleVariableDeclaration, ruleForEachStatement);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        {
+            get
+            {
+                return s_supportedRules;
+            }
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(syntaxContext =>
+            {
+                var node = (VariableDeclarationSyntax)syntaxContext.Node; 
+                if (node.Type.IsVar && !IsAnonymousType(node.Variables.Single(), syntaxContext.SemanticModel, syntaxContext.CancellationToken))
```

Comment:
Add comment to explain that this is ok because `var` can't have multiple variable dedclarators.